[th/deprecated-logging-warn] cda: use logger.warning() instead of deprecated warn()

thom311 opened the pull request:
Python's logging.warn() is long deprecated for logging.warning(). See for example [1].

This causes deprecation warnings like
```
  /root/cluster-deployment-automation/cda.py:22: DeprecationWarning: The 'warn' method is deprecated, use 'warning' instead
    logger.warn("Applying workaround for assisted installer issue")
  2024-03-08 12:11:14 WARNING: Applying workaround for assisted installer issue
```
[1] https://docs.python.org/3/library/logging.html#logging.warning

Zinkelburger commented:
There is a merge conflict because NodeConfig changed, and I removed the warning message by accident.
If you want to add the warning message back to the new code, in NodeConfig we could put
```python
if self.type:
    logger.warning("Deprecated 'type' in node config. Use 'kind' instead")
    self.kind = self.type
```

thom311 commented:
> There is a merge conflict because NodeConfig changed, and I removed the warning message by accident. If you want to add the warning message back to the new code, in NodeConfig we could put

Hi. I rebased the branch. There is now a new commit to restore the warning.

Zinkelburger reviewed (approved):
Looks good to me
